dessia_common/workflow/core.py

```diff
@@ -1941,6 +1943,138 @@ def _method_jsonschemas(self):
+                inst = f"{class_name}('Put here your object')"
+                instances += inst + ", "
+            instances += ']'
+            return instances
+
+        return f"{class_name}('Put here your object')"
+
+    @staticmethod
+    def get_files(type_files):
+        """ Generates file instances from a list of file types. """
+        file_class = type_files[0].__class__.__name__
+        if len(type_files) > 1:
+            instances = "["
+            for _ in range(len(type_files)):
+                instances += f"{file_class}.from_file('Insert your path here')" + ", "
+            instances += ']'
+            return instances
+        return f"{file_class}.from_file('Insert your path here')"
+
+    @staticmethod
+    def is_object_sequence(seq):
+        """ Checks if a sequence contains serializable objects. """
+        return any(isinstance(element, SerializableObject) for element in seq)
+
+    @staticmethod
+    def is_file_sequence(sequence):
+        """ Checks if a sequence contains binary files. """
+        return any(isinstance(element, BinaryFile) for element in sequence)
+
+    def get_sequence(self, sequence):
+        """ Processes a sequence based on its content. """
+        if self.is_object_sequence(sequence):
+            value = self.instantiate_objects(sequence)
+        elif self.is_file_sequence(sequence):
+            value = self.get_files(sequence)
+        else:
+            value = repr(sequence)
+        return value
+
+    def _compute_input_values(self):
+        """ Computes the input values for blocks and non-block variables. """
+        values_nonblock_variables = []
+        values_blocks = []
+        for i, variable in enumerate(self.workflow.inputs):
+            if variable in self.workflow.nonblock_variables:
+                values_nonblock_variables.append(self.input_values[i])
+            else:
+                values_blocks.append(self.input_values[i])
+
+        return values_nonblock_variables, values_blocks
+
+    def _to_script(self, workflow_script):
+        """ Computes elements for a to_script interpretation. """
+        input_info = {
+            "input_str": "",
+            "default_value": "",
+            "add_import": [],
+        }
+
+        block_input = "block_{}.inputs[{}]"
+
+        values_nonblock_variables, values_blocks = self._compute_input_values()
+
+        # Blocks
+        index_ = 0
+        for j, block in enumerate(self.workflow.blocks):
+            for i, input_ in enumerate(block.inputs):
+                input_key = block_input.format(j, i)
+                if input_key not in workflow_script.declaration:
+                    input_info["input_str"] += f"    workflow.input_index({input_key}): value_{j}_{i},\n"
+                    input_info["default_value"] += self._generate_default_value(values_blocks[index_], j, i)
+                    index_ += 1
+                    input_info["add_import"].extend(get_schema(annotation=input_.type_,
```

Comment:
Please simplify this line by assigning variables to schema and eventually imports

Reply:
I did as you asked, but I had a CI problem, that's why I encapsulated everything !
I will update  !

Reply:
done !